content/docs/2.6/scalers/predictkube.md

```diff
@@ -0,0 +1,96 @@
++++
+title = "Predictkube"
+availability = "v1.0+"
+maintainer = "Dysnix"
+description = "AI-based predictive scaling based on Prometheus metrics."
+layout = "scaler"
+go_file = "predictkube_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `predictkube` trigger that scales based on a predicting load based on `prometheus` metrics.
+
+```yaml
+triggers:
+- type: predictkube
+  metadata:
+    # Required fields:
+    predictHorizon: "2h"
+    historyTimeWindow: "7d"
+    prometheusAddress: http://<prometheus-host>:9090
+    metricName: http_requests_total
+    query: sum(rate(http_requests_total{deployment="my-deployment"}[2m]))
+    queryStep: "2m"
+    threshold: '100'
+```
+
+**Parameter list:**
+
+- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
+- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
+- `prometheusAddress` - Address of Prometheus server.
+- `metricName` - Name to identify the Metric in the external.metrics.k8s.io API. API. If using more than one trigger it is required that all `metricName`(s) be unique.
+- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
+- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
+- `threshold` - Value to start scaling for.
+
+### Authentication Parameters
+
+Predictkube Scaler supports one type of authentication - authentication by API key.
+
+**Auth gateway based authentication:**
+
+- `apiKey` - API key previously issued for this tenant.
```

Comment:
API key of what? :)

Reply:
for PredictKube API. It can be obtained for free at https://predictkube.com/

Reply:
By now I've figured that out, but this is something that should be as part of the documentation.

Can you add that please?

Reply:
Sure, we have added comment in https://github.com/dysnix/keda-docs/commit/1ff8f1dc02114b98a7c301f7025e05c33830ecb7

Thank you!